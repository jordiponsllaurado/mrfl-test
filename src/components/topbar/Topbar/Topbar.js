import PropTypes from 'prop-types';
import styled from 'styled-components';
import BurgerMenu from '../../BurgerMenu';
import Header from '../Header';
import SectionMenu from './SectionMenu';

const TopbarWrapper = styled.div`
  ${({ gradient }) => gradient && `background: ${gradient};`}
  ${({ color }) => color && `background: ${color};`}
`;

const Topbar = ({
  customizations,
  sections,
  sectionActive,
  onChangeSection,
  sectionVisible,
  headerVisible,
  onBurgerClick
}) => {
  const { background, sectionMenu, burgerMenu } = customizations;

  if (!headerVisible) {
    return (
      <div className="topbar">
        <BurgerMenu
          background={background.color}
          burgerMenu={burgerMenu.color}
          onBurgerClick={onBurgerClick}
        />
      </div>
    );
  }

  return (
    <TopbarWrapper className="topbar" color={background.color} gradient={background.gradient}>
      <Header onBurgerClick={onBurgerClick} {...customizations} />
      {sectionVisible && (
        <SectionMenu
          sections={sections}
          sectionActive={sectionActive}
          onChange={onChangeSection}
          color={sectionMenu.color}
          textSize={sectionMenu.textSize}
        />
      )}
    </TopbarWrapper>
  );
};

Topbar.propTypes = {
  customizations: PropTypes.shape({
    logo: PropTypes.shape({
      image: PropTypes.string,
      size: PropTypes.string
    }),
    background: PropTypes.shape({
      color: PropTypes.string,
      gradient: PropTypes.string
    }),
    burgerMenu: PropTypes.shape({
      color: PropTypes.string
    }),
    sectionMenu: PropTypes.shape({
      color: PropTypes.string,
      textSize: PropTypes.string
    })
  }),
  sections: PropTypes.arrayOf(PropTypes.string).isRequired,
  sectionActive: PropTypes.string.isRequired,
  onChangeSection: PropTypes.func.isRequired,
  sectionVisible: PropTypes.bool,
  headerVisible: PropTypes.bool,
  onBurgerClick: PropTypes.func
};

Topbar.defaultProps = {
  customizations: {
    logo: {
      image: 'https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg',
      size: '40px'
    },
    background: {
      color: 'white'
    },
    burgerMenu: {
      color: 'orange'
    },
    sectionMenu: {
      color: '',
      textSize: '22px'
    }
  },
  sectionVisible: true,
  headerVisible: true,
  onBurgerClick: null
};
export default Topbar;
