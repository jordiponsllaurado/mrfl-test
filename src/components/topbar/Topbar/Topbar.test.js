import { render } from '@testing-library/react';
import Topbar from './Topbar';

test('renders Topbar', () => {
  const { container } = render(<Topbar />);
  expect(container).toMatchSnapshot();
});

test('renders Topbar with customizations', () => {
  const customizations = {
    logo: {
      image: 'imageUrl',
      size: '30px'
    },
    bacground: {
      color: 'green'
    },
    burgerMenu: {
      color: 'black'
    }
  };
  const { container } = render(<Topbar {...customizations} />);
  expect(container).toMatchSnapshot();
});
