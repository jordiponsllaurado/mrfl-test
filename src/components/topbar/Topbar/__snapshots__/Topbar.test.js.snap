// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Topbar 1`] = `
<div>
  <div
    class="sc-dlfnbm eGszKx topbar"
    color="white"
  >
    <div
      class="header"
    >
      <div
        class="burger-menu"
        role="button"
        tabindex="0"
      >
        <span
          class="fa-layers fa-fw fa-2x"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-lg "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: white;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 fa-sm "
            data-icon="bars"
            data-prefix="fas"
            data-qa="burger-menu-icon"
            focusable="false"
            role="img"
            style="color: orange;"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <img
        alt="Marfeel"
        class="sc-bdfBwQ fkduWl"
        data-qa="logo"
        src="https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg"
      />
    </div>
    <div
      class="section-menu"
    >
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        facilitate
      </a>
      <a
        class="sc-gsTCUz fwxwnb section-item active"
        color=""
      >
        administration
      </a>
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        invoice
      </a>
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        bandwidth
      </a>
    </div>
  </div>
</div>
`;

exports[`renders Topbar with customizations 1`] = `
<div>
  <div
    class="sc-dlfnbm eGszKx topbar"
    color="white"
  >
    <div
      class="header"
    >
      <div
        class="burger-menu"
        role="button"
        tabindex="0"
      >
        <span
          class="fa-layers fa-fw fa-2x"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle fa-w-16 fa-lg "
            data-icon="circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style="color: white;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-bars fa-w-14 fa-sm "
            data-icon="bars"
            data-prefix="fas"
            data-qa="burger-menu-icon"
            focusable="false"
            role="img"
            style="color: orange;"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
              fill="currentColor"
            />
          </svg>
        </span>
      </div>
      <img
        alt="Marfeel"
        class="sc-bdfBwQ fkduWl"
        data-qa="logo"
        src="https://www.marfeel.com/public/images/landing/header/logo-marfeel.svg"
      />
    </div>
    <div
      class="section-menu"
    >
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        facilitate
      </a>
      <a
        class="sc-gsTCUz fwxwnb section-item active"
        color=""
      >
        administration
      </a>
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        invoice
      </a>
      <a
        class="sc-gsTCUz bRxRTt section-item"
        color=""
      >
        bandwidth
      </a>
    </div>
  </div>
</div>
`;
